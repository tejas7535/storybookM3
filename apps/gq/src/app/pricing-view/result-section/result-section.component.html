<p>result-section works!</p>
