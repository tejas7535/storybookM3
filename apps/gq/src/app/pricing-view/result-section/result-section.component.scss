@import 'libs/shared/ui/styles/src/lib/colors';
@import 'libs/shared/ui/styles/src/lib/fonts';

:host {
  height: 100%;
  width: 100%;

  .table-container {
    height: 100%;
    width: 100%;

    ag-grid-angular {
      height: 100%;
      min-height: auto;
      width: 100%;
    }
  }
}

.column-header {
  color: $schaeffler-carbon-gray;
  font-weight: 500;
}
