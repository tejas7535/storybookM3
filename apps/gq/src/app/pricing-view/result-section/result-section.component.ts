import { Component } from '@angular/core';

import {
  ClientSideRowModelModule,
  ColDef,
} from '@ag-grid-community/all-modules';
import {
  ColumnsToolPanelModule,
  FiltersToolPanelModule,
  MenuModule,
  RowGroupingModule,
  SideBarModule,
} from '@ag-grid-enterprise/all-modules';
import { translate } from '@ngneat/transloco';

@Component({
  selector: 'gq-result-section',
  templateUrl: './result-section.component.html',
  styleUrls: ['./result-section.component.scss'],
})
export class ResultSectionComponent {
  modules: any[] = [
    ClientSideRowModelModule,
    RowGroupingModule,
    FiltersToolPanelModule,
    ColumnsToolPanelModule,
    MenuModule,
    SideBarModule,
  ];

  defaultColumnDefs: ColDef = {
    sortable: true,
    filter: true,
    resizable: true,
    enablePivot: false,
    enableRowGroup: true,
    floatingFilter: true,
    menuTabs: ['generalMenuTab'],
  };

  columnDefs: any = [
    {
      headerName: translate(
        'pricingView.resultSection.tableHeader.generalInformation'
      ),
      children: [
        {
          headerName: translate('pricingView.resultSection.tableHeader.number'),
          field: 'nr',
          checkboxSelection: true,
        },
        {
          headerName: translate(
            'pricingView.resultSection.tableHeader.quotationNumber'
          ),
          field: 'qnr',
        },
        {
          headerName: translate(
            'pricingView.resultSection.tableHeader.positionNumber'
          ),
          field: 'posnr',
        },
        {
          headerName: translate(
            'pricingView.resultSection.tableHeader.customerID'
          ),
          field: 'cusid',
        },
        {
          headerName: translate(
            'pricingView.resultSection.tableHeader.customer'
          ),
          field: 'customer',
        },
      ],
    },
    {
      headerName: 'Customer Information',
      children: [
        {
          headerName: 'Cust. Segm (A,B,C)',
          field: 'cusseg',
        },
        {
          headerName: 'Incoterms',
          field: 'incoterms',
        },
        {
          headerName: 'Payment Terms',
          field: 'payterms',
        },
        {
          headerName: 'Keyaccount ID',
          field: 'keyid',
        },
        {
          headerName: 'Keyaccount',
          field: 'keyacc',
        },
        {
          headerName: 'Sub Keyaccount',
          field: 'subkey',
        },
      ],
    },
  ];

  rowData: any = [
    {
      nr: '1',
      qnr: '20090040',
      posnr: 10,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '2',
      qnr: '20090040',
      posnr: 11,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '3',
      qnr: '20090040',
      posnr: 12,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '4',
      qnr: '20090040',
      posnr: 13,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '5',
      qnr: '20090040',
      posnr: 14,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '6',
      qnr: '20090040',
      posnr: 15,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '7',
      qnr: '20090040',
      posnr: 16,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '8',
      qnr: '20090040',
      posnr: 17,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '9',
      qnr: '20090040',
      posnr: 18,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '10',
      qnr: '20090040',
      posnr: 19,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '11',
      qnr: '20090040',
      posnr: 20,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '12',
      qnr: '20090040',
      posnr: 21,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '13',
      qnr: '20090040',
      posnr: 22,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '14',
      qnr: '20090040',
      posnr: 23,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '15',
      qnr: '20090040',
      posnr: 10,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '16',
      qnr: '20090040',
      posnr: 10,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '17',
      qnr: '20090040',
      posnr: 10,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '18',
      qnr: '20090040',
      posnr: 10,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '19',
      qnr: '20090040',
      posnr: 10,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '75 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
    {
      nr: '20',
      qnr: '20090040',
      posnr: 10,
      cusid: '26605',
      customer: 'ABB OY',
      cusseg: 'ABB QY',
      incoterms: 'DAP',
      payterms: '30 days',
      keyid: '70',
      keyacc: 'ABB',
      subkey: '10737',
    },
  ];

  sideBar: any = {
    toolPanels: [
      {
        id: 'columns',
        labelDefault: 'Columns',
        labelKey: 'columns',
        iconKey: 'columns',
        toolPanel: 'agColumnsToolPanel',
        toolPanelParams: {
          suppressValues: true,
          suppressPivots: true,
          suppressPivotMode: true,
        },
      },
    ],
    position: 'right',
  };
}
