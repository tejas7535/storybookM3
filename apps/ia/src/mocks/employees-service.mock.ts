import { Employee } from '../app/shared/models';

export const ROOT = ({
  employeeId: '123',
  parentEmployeeId: undefined,
  exitDate: undefined,
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const LEVEL_2_EMPLOYEE_A = ({
  employeeId: '456',
  parentEmployeeId: '123',
  exitDate: undefined,
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const LEVEL_2_EMPLOYEE_B = ({
  employeeId: '789',
  parentEmployeeId: '123',
  exitDate: undefined,
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const LEVEL_2_EMPLOYEE_C = ({
  employeeId: '13',
  parentEmployeeId: '123',
  exitDate: '2015-10-10',
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
  entryDate: '2010-10-10',
  terminationDate: '2015-08-10',
} as unknown) as Employee;

export const LEVEL_3_EMPLOYEE_A = ({
  employeeId: '33',
  parentEmployeeId: '789',
  exitDate: '2015-10-10',
  totalSubordinates: 0,
  directSubordinates: 0,
  directAttrition: 0,
  totalAttrition: 0,
} as unknown) as Employee;

export const EMPLOYEES: Employee[] = [
  ROOT,
  LEVEL_2_EMPLOYEE_A,
  LEVEL_2_EMPLOYEE_B,
  LEVEL_2_EMPLOYEE_C,
  LEVEL_3_EMPLOYEE_A,
];

const map = new Map<string, Employee[]>();
map.set('-1', [ROOT]);
map.set('123', [LEVEL_2_EMPLOYEE_A, LEVEL_2_EMPLOYEE_B, LEVEL_2_EMPLOYEE_C]);
map.set('789', [LEVEL_3_EMPLOYEE_A]);

export const EMPLOYEE_MAP = map;
