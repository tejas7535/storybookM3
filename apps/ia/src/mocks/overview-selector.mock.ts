import {
  Employee,
  EmployeeAttritionMeta,
  LeavingType,
} from '../app/shared/models';

export const EMPLOYEES_FOR_SELECTORS: Employee[] = [
  ({
    employeeId: '123',
    parentEmployeeId: undefined,
    orgUnit: 'Schaeffler_IT',
    directSubordinates: 3,
    totalSubordinates: 4,
    directAttrition: 1,
    totalAttrition: 2,
    exitDate: undefined,
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '456',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 0,
    totalAttrition: 0,
    exitDate: new Date('2020-02-28'),
    reasonForLeaving: LeavingType.FORCED,
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '678',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 1,
    totalSubordinates: 1,
    directAttrition: 1,
    totalAttrition: 1,
    exitDate: undefined,
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '999',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 0,
    totalAttrition: 0,
    exitDate: undefined,
    entryDate: new Date('2020-06-06'),
    directLeafChildren: [],
  } as unknown) as Employee,
  ({
    employeeId: '91011',
    parentEmployeeId: '678',
    orgUnit: 'Schaeffler_IT_1_2',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 0,
    totalAttrition: 0,
    exitDate: new Date('2020-05-05'),
    reasonForLeaving: LeavingType.UNFORCED,
    directLeafChildren: [],
  } as unknown) as Employee,
];

export const EXPECTED_FILTERED_EMPLOYEES: Employee[] = [
  ({
    employeeId: '123',
    exitDate: undefined,
    parentEmployeeId: undefined,
    orgUnit: 'Schaeffler_IT',
    directSubordinates: 2,
    totalSubordinates: 2,
    directAttrition: 1,
    totalAttrition: 2,
    directLeafChildren: [
      ({
        employeeId: '999',
        parentEmployeeId: '123',
        orgUnit: 'Schaeffler_IT_1',
        directSubordinates: 0,
        totalSubordinates: 0,
        directAttrition: 0,
        totalAttrition: 0,
        entryDate: new Date('2020-06-06'),
        exitDate: undefined,
        directLeafChildren: [],
        attritionMeta: new EmployeeAttritionMeta(
          'Schaeffler_IT_1',
          0,
          0,
          0,
          1,
          1,
          0,
          1,
          0
        ),
      } as unknown) as Employee,
    ],
    attritionMeta: new EmployeeAttritionMeta(
      'Schaeffler_IT',
      45.28,
      2,
      0,
      1,
      1,
      0,
      1,
      0
    ),
  } as unknown) as Employee,
  ({
    employeeId: '678',
    parentEmployeeId: '123',
    orgUnit: 'Schaeffler_IT_1',
    directSubordinates: 0,
    totalSubordinates: 0,
    directAttrition: 1,
    totalAttrition: 1,
    directLeafChildren: [],
    exitDate: undefined,
    attritionMeta: new EmployeeAttritionMeta(
      'Schaeffler_IT_1',
      29.27,
      1,
      0,
      1,
      1,
      0,
      1,
      0
    ),
  } as unknown) as Employee,
];
