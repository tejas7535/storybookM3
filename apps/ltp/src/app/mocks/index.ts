export * from 'mock.constants';
export * from 'mock.service';
