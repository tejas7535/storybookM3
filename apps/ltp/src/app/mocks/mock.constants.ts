import { EChartsOption } from 'echarts';

import { LTPState } from '../core/store';
import { initialState } from '../core/store/reducers/input.reducer';
import { PredictionState } from '../core/store/reducers/prediction.reducer';
import {
  BurdeningType,
  Material,
  Prediction,
  PredictionResult,
} from '../shared/models';

export const mockedMaterials: Material[] = [
  { name: 'Plastik', heatTreatment: 'volleHitze', hardness: 90001 },
  { name: 'Holz', heatTreatment: 'halbeHitze', hardness: 1 },
];

export const mockedPredictions: Prediction[] = [
  { id: 0, name: 'Arnold Strong' },
  { id: 1, name: 'Silvester Stallone' },
];

export const mockedBurdeningTypes: BurdeningType[] = [
  { id: 0, name: 'Jet Lee' },
  { id: 1, name: 'Tyson Fury' },
];

export const mockedLoadsResult = {
  loads: {
    x: [96.0, 214.0],
    y: [1050.0, 1000.0],
  },
  woehler: {
    case: 1,
    fatigue_strength: 129.60351058284022,
    line_percentile_1: [
      [
        [1158691.4165629777, 129.60351058284027],
        [10000, 301.21165055865276],
      ],
      [
        [1158691.4165629777, 129.60351058284027],
        [10000000, 129.60351058284027],
      ],
    ],
    line_percentile_10: [
      [
        [1158691.4165629777, 129.60351058284027],
        [10000, 301.21165055865276],
      ],
      [
        [1158691.4165629777, 129.60351058284027],
        [10000000, 129.60351058284027],
      ],
    ],
    line_percentile_50: [
      [
        [1158691.4165629777, 129.60351058284027],
        [460.9834219198255, 519.9999999999998],
      ],
      [
        [1158691.4165629777, 129.60351058284027],
        [10000000, 129.60351058284027],
      ],
    ],
    line_percentile_90: [
      [
        [1158691.4165629777, 129.60351058284027],
        [10000, 301.21165055865276],
      ],
      [
        [1158691.4165629777, 129.60351058284027],
        [10000000, 129.60351058284027],
      ],
    ],
    line_percentile_99: [
      [
        [1158691.4165629777, 129.60351058284027],
        [10000, 301.21165055865276],
      ],
      [
        [1158691.4165629777, 129.60351058284027],
        [10000000, 129.60351058284027],
      ],
    ],
    percentile_1: 2022,
    percentile_10: 2022,
    percentile_50: 2022,
    percentile_90: 2022,
    percentile_99: 2022,
    plot1: [
      [10000, 400],
      [2022, 400],
      [2022, 103.68280846627218],
      [2022, 400],
    ],
    plot2: [
      [10000, 301.21165055865265],
      [460.9834219198255, 520],
      [1158691.4165629777, 129.60351058284022],
      [10000000, 129.60351058284022],
    ],
    slope: 5.635329994802062,
    stress_amplitude: 400,
    x0: 460.9834219198255,
    xlim0: 10000,
    xlim1: 10000000,
    y0: 520,
  },
};

export const mockedPredictionResult: PredictionResult = {
  woehler: {
    snCurve: {
      '0': {
        x: 460.9834219198255,
        y: 520,
      },
      '1': {
        x: 1158691.4165629777,
        y: 129.60351058284022,
      },
      '2': {
        x: 10000000,
        y: 129.60351058284022,
      },
    },
    snCurveLow: {
      '0': {
        x: 460.9834219198255,
        y: 520,
      },
      '1': {
        x: 1158691.4165629777,
        y: 129.60351058284022,
      },
      '2': {
        x: 10000000,
        y: 129.60351058284022,
      },
    },
    snCurveHigh: {
      '0': {
        x: 460.9834219198255,
        y: 520,
      },
      '1': {
        x: 1158691.4165629777,
        y: 129.60351058284022,
      },
      '2': {
        x: 10000000,
        y: 129.60351058284022,
      },
    },
    appliedStress: {
      '0': {
        x: 2022,
        y: 400,
      },
      '1': {
        x: 2022,
        y: 103.68280846627218,
      },
      '2': {
        x: 10000,
        y: 400,
      },
    },
    percentile1: {
      '0': {
        x: 460.9834219198255,
        y: 519.9999999999998,
      },
      '1': {
        x: 1158691.4165629777,
        y: 129.60351058284027,
      },
      '2': {
        x: 10000000,
        y: 129.60351058284027,
      },
    },
    percentile10: {
      '0': {
        x: 460.9834219198255,
        y: 519.9999999999998,
      },
      '1': {
        x: 1158691.4165629777,
        y: 129.60351058284027,
      },
      '2': {
        x: 10000000,
        y: 129.60351058284027,
      },
    },
    percentile90: {
      '0': {
        x: 460.9834219198255,
        y: 519.9999999999998,
      },
      '1': {
        x: 1158691.4165629777,
        y: 129.60351058284027,
      },
      '2': {
        x: 10000000,
        y: 129.60351058284027,
      },
    },
    percentile99: {
      '0': {
        x: 460.9834219198255,
        y: 519.9999999999998,
      },
      '1': {
        x: 1158691.4165629777,
        y: 129.60351058284027,
      },
      '2': {
        x: 10000000,
        y: 129.60351058284027,
      },
    },
  },
  haigh: {
    snCurve: {
      '0': {
        x: 0,
        y: 129.60351058284022,
      },
      '1': {
        x: 127.48671337634478,
        y: 127.48671337634478,
      },
    },
    appliedStress: {
      '0': {
        x: 0,
        y: 0,
      },
      '1': {
        x: 229.60351058284022,
        y: 229.60351058284022,
      },
    },
  },
  kpi: {
    fatigue: {
      '0': 127.48671337634478,
      '1': 129.60351058284022,
    },
    slope: 5.635329994802062,
  },
};

export const mockedStatisticalResult = {
  woehler: {
    analytical: {
      sa_fkm: 267.29999999999995,
      sa_murakami: 357.89106194324967,
    },
    statistical_sn_curve: {
      percentile_50: {
        '0': { x: 10000, y: 255.3071591874725 },
        '1': { x: 329718.17701544514, y: 122.46499412533268 },
        '2': { x: 10000000, y: 122.46499412533268 },
      },
      percentile_10: {
        '0': { x: 10000, y: 342.9400049487193 },
        '1': { x: 329718.17701544514, y: 164.50046220814025 },
        '2': { x: 10000000, y: 164.50046220814025 },
      },
      percentile_90: {
        '0': { x: 10000, y: 190.06748874959695 },
        '1': { x: 329718.17701544514, y: 91.1710191254138 },
        '2': { x: 10000000, y: 91.1710191254138 },
      },
    },
  },
  haigh: {
    analytical: {
      r0_fkm: 267.29999999999995,
      r1_fkm: 241.26726238830219,
      r0_murakami: 357.89106194324967,
      r1_murakami: 302.20353030591974,
    },
    statistical: { r0: 108.09316567203312, r1: 122.46499412533268 },
  },
};

export const mockedPredictionRequestWithKpi: PredictionState = {
  predictionRequest: {
    prediction: 0,
    mpa: 400,
    v90: 0,
    hv: 180,
    hv_lower: 180,
    hv_upper: 180,
    rrelation: -1,
    burdeningType: 0,
    model: 5,
    spreading: 0,
    rArea: 5,
    es: 0,
    rz: 0,
    hv_core: 500,
    a90: 100,
    gradient: 1,
    multiaxiality: 0,
  },
  statisticalRequest: {
    rz: 0,
    es: 0,
    hardness: 180,
    r: -1,
    rArea: 5,
    v90: 1,
    loadingType: 0,
  },
  statisticalResult: {
    woehler: {
      analytical: {
        sa_fkm: 67.29999999999995,
        sa_murakami: 293.89106194324967,
      },
      statistical_sn_curve: {
        percentile_50: {
          0: { x: 10000, y: 255.3071591874725 },
          1: { x: 329718.17701544514, y: 122.46499412533268 },
          2: { x: 10000000, y: 122.46499412533268 },
        },
        percentile_10: {
          0: { x: 10000, y: 342.9400049487193 },
          1: { x: 329718.17701544514, y: 164.50046220814025 },
          2: { x: 10000000, y: 164.50046220814025 },
        },
        percentile_90: {
          0: { x: 10000, y: 190.06748874959695 },
          1: { x: 329718.17701544514, y: 91.1710191254138 },
          2: { x: 10000000, y: 91.1710191254138 },
        },
      },
    },
    haigh: {
      analytical: {
        r0_fkm: 267.29999999999995,
        r1_fkm: 241.26726238830219,
        r0_murakami: 357.89106194324967,
        r1_murakami: 302.20353030591974,
      },
      statistical: { r0: 108.09316567203312, r1: 122.46499412533268 },
    },
  },
  predictionResult: {
    haigh: undefined,
    woehler: undefined,
    kpi: {
      fatigue: {
        0: 10,
        1: 20,
      },
      slope: 120,
    },
  },
  loadsRequest: {
    data: [0, 1, 2, 3],
    status: 1,
    error: undefined,
    conversionFactor: 1,
    repetitionFactor: 1,
    method: 'FKM',
  },
  loads: undefined,
};

export const mockedPredictionRequestWithLimits: LTPState = {
  input: {
    ...initialState,
    display: {
      ...initialState.display,
      showFKM: true,
      showMurakami: true,
      showStatistical: true,
    },
  },
  prediction: {
    ...mockedPredictionRequestWithKpi,
    predictionResult: {
      ...mockedPredictionResult,
    },
  },
};

export const mockedPredictionResultGraphData: EChartsOption = {
  dataset: {
    source: [
      {
        x: 10000,
        y7: 169.04995684059463,
      },
      {
        x: 1000000,
        y7: 67.29999999999995,
      },
      {
        x: 10000000,
        y7: 67.29999999999995,
      },
      {
        x: 10000,
        y8: 738.2209708371904,
      },
      {
        x: 1000000,
        y8: 293.89106194324967,
      },
      {
        x: 10000000,
        y8: 293.89106194324967,
      },
      {
        x: 10000,
        y9: 255.3071591874725,
      },
      {
        x: 329718.17701544514,
        y9: 122.46499412533268,
      },
      {
        x: 10000000,
        y9: 122.46499412533268,
      },
      {
        x: 10000000,
        yHigh: 129.60351058284022,
        yLow: 129.60351058284022,
      },
      {
        x: 1158691.4165629777,
        yHigh: 129.60351058284022,
        yLow: 129.60351058284022,
      },
      {
        x: 10000,
        y2: 301.21165055865265,
      },
      {
        x: 1158691.4165629777,
        y2: 129.60351058284022,
      },
      {
        x: 10000000,
        y2: 129.60351058284022,
      },
      {
        x: 10000,
        y3: 301.21165055865254,
      },
      {
        x: 1158691.4165629777,
        y3: 129.60351058284027,
      },
      {
        x: 10000000,
        y3: 129.60351058284027,
      },
      {
        x: 10000,
        y5: 301.21165055865254,
      },
      {
        x: 1158691.4165629777,
        y5: 129.60351058284027,
      },
      {
        x: 10000000,
        y5: 129.60351058284027,
      },
      {
        x: 10000,
        y6: 301.21165055865254,
      },
      {
        x: 1158691.4165629777,
        y6: 129.60351058284027,
      },
      {
        x: 10000000,
        y6: 129.60351058284027,
      },
      {
        x: 10000,
        y4: 301.21165055865254,
      },
      {
        x: 1158691.4165629777,
        y4: 129.60351058284027,
      },
      {
        x: 10000000,
        y4: 129.60351058284027,
      },
    ],
  },
  xAxis: {
    max: 10000000,
    min: 10000,
  },
  yAxis: {
    max: 812.0430679209095,
    min: 60.56999999999996,
  },
  // tslint:disable-next-line: max-file-line-count
};
