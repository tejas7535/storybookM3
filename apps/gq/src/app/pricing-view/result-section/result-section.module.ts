import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { ResultSectionComponent } from './result-section.component';

@NgModule({
  declarations: [ResultSectionComponent],
  imports: [CommonModule],
  exports: [ResultSectionComponent],
})
export class ResultSectionModule {}
