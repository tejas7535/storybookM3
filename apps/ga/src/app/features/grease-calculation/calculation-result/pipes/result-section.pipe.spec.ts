import { greaseResultMock } from '@ga/testing/mocks';

import { ResultSectionPipe } from './result-section.pipe';

jest.mock('@jsverse/transloco', () => ({
  translate: jest.fn((key: string) => key),
}));

describe('ResultSectionPipe', () => {
  let pipe: ResultSectionPipe;

  beforeEach(() => {
    pipe = new ResultSectionPipe();
  });

  it('should tranform the result matching the snapshot', () => {
    expect(pipe).toBeTruthy();
  });

  describe('transform', () => {
    it('should transform the result', () => {
      const result = pipe.transform(greaseResultMock);

      expect(result).toMatchSnapshot();
    });
  });

  describe('splitBadgeText', () => {
    it.each([
      [
        '<span>text</span>',
        { badgeText: 'text', badgeSecondaryText: undefined },
      ],
      [
        '<span>text</span> <span>secondary</span>',
        { badgeText: 'text', badgeSecondaryText: 'secondary' },
      ],
      [
        '<span>text</span><br><span class="something">secondary</span>',
        { badgeText: 'text', badgeSecondaryText: 'secondary' },
      ],
      ['text', { badgeText: 'text', badgeSecondaryText: undefined }],
      ['', { badgeText: undefined, badgeSecondaryText: undefined }],
    ])('should split badge text "%s" into %j', (input, expected) => {
      const result = pipe['splitBadgeText'](input);
      expect(result).toEqual(expected);
    });
  });

  describe('getKappaBadgeColorClass', () => {
    it.each([
      ['0,0', 'bg-error-container text-error'],
      ['-1', 'bg-error-container text-error'],
      ['0.9', 'bg-error-container text-error'],
      ['4,1', 'bg-error-container text-error'],
      ['4,0', 'bg-success-container text-success'],
      ['1.5', 'bg-success-container text-success'],
      ['1,5', 'bg-success-container text-success'],
      ['3,9', 'bg-success-container text-success'],
    ])(
      'should return the right formatting class for %s the kappa badge',
      (input, expected) => {
        const result = pipe['getKappaBadgeColorClass'](input);
        expect(result).toEqual(expected);
      }
    );
  });
});
