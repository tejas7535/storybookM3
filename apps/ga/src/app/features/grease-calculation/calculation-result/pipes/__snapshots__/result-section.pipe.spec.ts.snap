// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultSectionPipe transform should transform the result 1`] = `
Object {
  "greaseSelection": Object {
    "badgeClass": "bg-surface-container",
    "badgeSecondaryText": undefined,
    "badgeText": "~ 51440 days",
    "concept1": false,
    "extendable": true,
    "labelValues": Array [
      Object {
        "label": "calculationResult.greaseServiceLife",
        "labelHint": undefined,
        "value": "~ 51440 days",
      },
      Object {
        "label": "calculationResult.baseOilViscosityAt40",
        "labelHint": undefined,
        "value": "1234567.89 mock_unit",
      },
      Object {
        "label": "calculationResult.lowerTemperatureLimit",
        "labelHint": "calculationResult.lowerTemperatureLimitTooltip",
        "value": "1234567.89 mock_unit",
      },
      Object {
        "label": "calculationResult.upperTemperatureLimit",
        "labelHint": "calculationResult.upperTemperatureLimitTooltip",
        "value": "1234567.89 mock_unit",
      },
      Object {
        "label": "calculationResult.density",
        "labelHint": undefined,
        "value": "1234567.89 mock_unit",
      },
      Object {
        "label": "calculationResult.h1Registration",
        "labelHint": undefined,
        "value": "1234567.89",
      },
    ],
    "mainValue": "greaseServiceLife",
    "title": "greaseSelection",
  },
  "initialLubrication": Object {
    "badgeClass": "bg-surface-container",
    "badgeSecondaryText": "1234567.9 mock_unit",
    "badgeText": "1234567.9 g",
    "concept1": false,
    "extendable": false,
    "mainValue": "initialGreaseQuantity",
    "title": "initialLubrication",
  },
  "performance": Object {
    "badgeClass": "bg-error-container text-error",
    "badgeSecondaryText": undefined,
    "badgeText": "1234567.89",
    "concept1": false,
    "extendable": true,
    "labelValues": Array [
      Object {
        "label": "calculationResult.viscosityRatio",
        "labelHint": undefined,
        "value": "1234567.89",
      },
      Object {
        "label": "calculationResult.additiveRequired",
        "labelHint": "calculationResult.additiveRequiredTooltip",
        "value": "1234567.89",
      },
      Object {
        "label": "calculationResult.lowFriction",
        "labelHint": undefined,
        "value": "0 (suitable)",
      },
      Object {
        "label": "calculationResult.suitableForVibrations",
        "labelHint": undefined,
        "value": "0 (suitable)",
      },
      Object {
        "label": "calculationResult.supportForSeals",
        "labelHint": undefined,
        "value": "0 (suitable)",
      },
    ],
    "mainValue": "viscosityRatio",
    "title": "performance",
  },
  "relubrication": Object {
    "badgeClass": "bg-surface-container",
    "badgeSecondaryText": "51440328.7 mock_unit/hours",
    "badgeText": "51440328.7 g/hours",
    "concept1": true,
    "extendable": true,
    "labelValues": Array [
      Object {
        "label": "calculationResult.relubricationQuantityPer1000OperatingHours",
        "labelHint": undefined,
        "value": "<span>51440328.7 g/hours</span><br><span class=\\"text-low-emphasis\\">51440328.7 mock_unit/hours</span>",
      },
      Object {
        "label": "calculationResult.relubricationPer365days",
        "labelHint": "calculationResult.relubricationQuantityPer365daysTooltip",
        "value": "<span>450617279.8 g/365 days</span><br><span class=\\"text-low-emphasis\\">450617279.8 mock_unit/365 days</span>",
      },
      Object {
        "label": "calculationResult.relubricationPer30days",
        "labelHint": undefined,
        "value": "<span>37037036.7 g/30 days</span><br><span class=\\"text-low-emphasis\\">37037036.7 mock_unit/30 days</span>",
      },
    ],
    "mainValue": "relubricationQuantityPer1000OperatingHours",
    "title": "relubrication",
  },
}
`;
